src/unix/solarish/mod.rs

```diff
@@ -557,6 +559,21 @@ cfg_if! {
             }
         }
 
+        impl ::fmt::Debug for door_desc_t__d_data {
+            fn fmt(&self, f: &mut ::fmt::Formatter) -> ::fmt::Result {
+                unsafe {
+                    match self {
+                        Self { d_desc } => f.debug_struct("door_desc_t__d_data")
+                            .field("d_desc", &self.d_desc)
+                            .finish(),
+                        Self { d_resv } => f.debug_struct("door_desc_t__d_data")
+                            .field("d_resv", &self.d_resv)
+                            .finish(),
+                    }
+                }
+            }
+        }
```

Comment:
The second match arm is unreachable (it will never get called), and it should be producing an "unreachable code" warning which should fail to build when warnings are turned into errors.

Reply:
Yeap, I see now, but I can't see where in the pattern match that I did it wrong. I tried to follow the `pattern-matching-on-unions` on the Reference book, but it looks like it didn't worked as expected.

Reply:
The first pattern will always match because all variants of unions always overlap - notice that, unlike enums, unions do not have a discriminant that tells you which variant is active. 

I don't think there is a meaningful way of implementing `Debug` for this union right now. The union members are public, and it is currently unclear whether it would be ok for Rust unions to be uninitialized, so  you might need extra context information to know what to print.

Reply:
So in that case, is it ok to allow `missing_debug_implementations` for that union?

Reply:
Yes!